src/perf_counters.cc

```diff
@@ -114,6 +114,50 @@ void PerfCounters::CloseCounters() const {
     close(fd);
   }
 }
+#elif defined(BENCHMARK_OS_MACOSX)
```

Comment:
i wonder if it's worth splitting this file into perf_counters_linux.cc and perf_counters_osx.cc

(then we can selectively compile on top of using these guards)

Reply:
not a bad idea